services/core-api/app/api/mines/alerts/resources/mine_alert.py

```diff
@@ -138,3 +138,31 @@ def delete(self, mine_guid, mine_alert_guid):
         alert.save()
 
         return ('', 204)
+
+
+class GlobalMineAlertListResource(Resource, UserMixin):
+
+    @api.doc(description='Retrive a list of alerts for all mines')
+    @api.marshal_with(PAGINATED_GLOBAL_MINE_ALERT_LIST, code=200)
+    @requires_role_view_all
+    def get(self):
+        parser = reqparse.RequestParser()
+        parser.add_argument(
+            'page', type=int, help='page for pagination', location='args', store_missing=False)
+        parser.add_argument(
+            'per_page', type=int, help='records per page', location='args', store_missing=False)
+        args = parser.parse_args()
+        page = args.get('page')
+        per_page = args.get('per_page') if args.get('per_page') else 10  # default per page is 10
+
+        records, pagination_details = MineAlert.find_all_mine_alerts(page, per_page)
+
+        if not records:
+            raise BadRequest('Unable to fetch global alerts.')
```

Comment:
Do we know it's a bad request though, if we don't get any results? I think BadRequest should probably be triggered if the page number they're requesting is greater than the total number of pages, negative number, etc, but otherwise I think we could just return [] for records. 

I'm thinking of a scenario where there's a local database with no mine alerts in the table, or if they're all deleted for some business reason, or if there were (hypothetical) filters that excluded all the results- what would we want to return then?

Reply:
As discussed, `filter_by` query will return an empty array in case there's no any records. Hence if checking on `not records` (`None`) is an unreachable code. So removed the `if` check.